src/language-server/settings.ts

```diff
@@ -47,6 +63,62 @@ export default class Settings extends AutoDisposable {
       this.latest = {...Settings.defaults, ...settings};
       fire({newer: this.latest, older: previous});
     });
+    this.fileSynchronizer = fileSynchronizer;
+
+    this._disposables.push(
+        this.fileSynchronizer.fileChanges.listen((changes) => {
+          for (const change of changes) {
+            const workspacePath = converter.getWorkspacePathToFile(change);
+            if (workspacePath === 'polymer.json') {
+              this.updateProjectConfig();
+            }
+          }
+        }));
+    this.updateProjectConfig();
+
+    const {fire: firePcc, stream: pccStream} =
+        EventStream.create<Change<ProjectConfigInfo>>();
+    this.projectConfigChangeStream = pccStream;
+    this.fireProjectConfigChange = firePcc;
+  }
+
+
+  private async updateProjectConfig() {
+    let jsonContent;
+    try {
+      jsonContent = await this.fileSynchronizer.urlLoader.load(
+          'polymer.json' as ResolvedUrl);
```

Comment:
here we expect to find `polymer.json` in the current folder, correct? I guess it's somewhat related to the comment above

Reply:
Yeah, same as above.